### ISSUE TITLE
Is Travis CI a better choice for us

### ISSUE THREAD

**Issue by matt-raffel-kiva** [kiva/protocol-architecture#3]
We have the potential we will have to make some changes to some parts of our CI, as some of the CI orbs will be deprecated in the next few weeks.  See [issue #2](https://github.com/kiva/protocol-architecture/issues/2).

This issue is intended to evaluate the feasibility of going to Travis CI in place of Circle CI.  Thoughts:

1. We could move only affected repos to travis and keep circle ci where is remains viable.
2. What were the reasons for using circle ci?  
3. or Stay with CircleCI for all of it

Outcome of this issue is either a) we may not use travis ci or b) we may use travis ci or c) we could use travis and rather wait until a later time to decide.

**Comment by voutasaurus** [comment]
I can speak to question 2 "What were the reasons for using circle ci?"

When we last assessed CI vendor the major decision was SaaS (circle/travis/wercker/github actions) vs on-prem (Jenkins). The argument in favor of the former was largely that the config syntax is much easier to understand in the modern SaaS offerings vs Jenkins. The argument in favor of Jenkins was that marketplace was already using it and we could share resources / expertise.

In my own personal experience the quality difference between circle/travis/wercker has been pretty minimal, they all work basically the same way. Github actions was relatively new when we decided on circle so we didn't give it much thought, although since then we've been using it more, especially for some frontend stuff.

tldr it wasn't so much that we decided on circle, we decided against Jenkins.

**Comment by voutasaurus** [comment]
My own take on this is that switching everything over to travis is probably a lot of busywork and wont achieve much.

On the other hand if we were to use travis for new public repos as a way to compare / contrast with circleci I wouldn't complain.

Generally my position on tools is that if somebody wants to use a different tool they are welcome to but they should expect to spend they own time figuring it out and supporting their tool of choice. At least until other team members have enough time to catch up with any subtleties of that tool.

**Comment by matt-raffel-kiva** [comment]
> My own take on this is that switching everything over to travis is probably a lot of busywork and wont achieve much.
> 
> On the other hand if we were to use travis for new public repos as a way to compare / contrast with circleci I wouldn't complain.
> 
> Generally my position on tools is that if somebody wants to use a different tool they are welcome to but they should expect to spend they own time figuring it out and supporting their tool of choice. At least until other team members have enough time to catch up with any subtleties of that tool.

All good points.  

The reason the idea came up is that someone in kiva (I forgot exactly who) demoed some CI changes and they were using travis.   The reason I wondered if its worth considering (and I should have noted this above) is:  "will using travis help us avoid pitfalls of Orbs being deprecated"?

**Comment by voutasaurus** [comment]
> "will using travis help us avoid pitfalls of Orbs being deprecated"?

This is a good question. Travis doesn't have "orbs", that's a circle specific term for a config script library. I believe that travis doesn't have anything that is directly comparable to the circleci orb repository. As far as I know it compensates for this by having more features directly built-in. That has the downside of less flexibility, which it compensates for by letting you write your own config scripts across multiple files (with circle everything is just in .circleci/config.yml).

Also, if you're referring to the deprecation of the ubuntu16 image, that's not an orb per-se, that's the virtual machine that the CI job runs on, and deprecation of those will also happen with Travis. One way to avoid this OS deprecation (in both circleci and travis) in general is to not use docker-compose in CI and instead use unit tests (we previously decided that using unit tests only would reduce the effectiveness and flexibility of our tests given how much of our system is based on comms between services).

**Comment by matt-raffel-kiva** [comment]
Im ready to close this issue.  The discussion above is good and is enough for me to say Travis isn't something we are interested in perusing at this time.  @voutasaurus
